igvc_gazebo/nodes/ground_truth/main.cpp

```diff
@@ -106,12 +107,16 @@ int main(int argc, char** argv)
   ros::NodeHandle pNh("~");
 
   std::string ground_truth_topic, estimate_topic, pub_topic, diff_topic;
-  igvc::param(pNh, "ground_truth_sub_topic", ground_truth_topic, std::string("/ground_truth/state_raw"));
-  igvc::param(pNh, "ground_truth_pub_topic", pub_topic, std::string("/ground_truth"));
+
+  using namespace assertions;
+  Asserter asserter;
+
+  asserter.param(pNh, "ground_truth_sub_topic", ground_truth_topic, std::string("/ground_truth/state_raw"));
+  asserter.param(pNh, "ground_truth_pub_topic", pub_topic, std::string("/ground_truth"));
```

Comment:
I guess the main decision here was that there were two main outcomes for a failed parameter assertions:
1. Call ros::shutdown, and have the node shutdown when it gets to `while (ros::ok())` or `ros::spin()` (What IGVC currently uses)
2. Call `std::exit(-1)` and have the node just exit from there (What RR currently uses)

I probably should have discussed this with @ebretl first but I changed this to a class so that when constructing `Asseorter` you could specify which behaviour was desired. If we agree on one behaviour though we could probably move this back to free functions.

Reply:
Even then, that's known at compile time and it seems like it makes more sense to either switch on a compilation flag or just have two separate interfaces that call out to the same backing implementation.

Reply:
Actually fk it I talked with @evan10s and he's ok with using `ros::shutdown` as well. I'm gonna move this back to a free function.